src/components/markdown/ProblemsList/ProblemStatusCheckbox.tsx

```diff
@@ -169,46 +175,268 @@ export default function ProblemStatusCheckbox({
+                                value="setStatus"
+                                onClick={event => setPrivate(true)}
+                              />
+                              <span className="ml-2">Public</span>
+                            </label>
+
+                            <label className="inline-flex items-center ml-6">
+                              <input
+                                type="radio"
+                                className="form-radio"
+                                name="setStatus"
+                                value="setStatus"
+                                onClick={event => setPrivate(false)}
+                              />
+                              <span className="ml-2">Private</span>
+                            </label>
+                          </div>
+                        </div>
+                        <div>
+                          <label
+                            htmlFor="about"
+                            className="block text-sm font-medium text-gray-700"
+                          >
+                            Solution Code
+                          </label>
+                          <div className="mt-1">
+                            <textarea
+                              id="about"
+                              name="about"
+                              rows="3"
+                              className="shadow-sm focus:ring-indigo-500 focus:border-indigo-500 mt-1 block w-full sm:text-sm border-gray-300 rounded-md"
+                              placeholder="What do you think about this problem?"
+                              onChange={event =>
+                                setsolutionCode(event.target.value)
+                              }
+                            >
+                              {' '}
+                            </textarea>
+                          </div>
+                        </div>
+                        <p className="mt-2 text-sm text-gray-500">
+                          This is a survey to collect user input on the provided
+                          problem
+                        </p>
+                      </div>
+
+                      <div className="px-4 py-3 bg-gray-50 text-right sm:px-6">
+                        <button
+                          type="submit"
+                          className="inline-flex justify-center py-2 px-4 border border-transparent shadow-sm text-sm font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
+                          onClick={e => handleSumbit(e, problem)}
+                        >
+                          Submit
+                        </button>
+                      </div>
+                    </div>
+                  </form>
+                </div>
+              </div>
+            </div>
+          </div>
+        </Transition>
+      </div>
+    </Transition>
+  );
```

Comment:
can this be refactored to another file?

Reply:
Can we do that later?

Reply:
no.... this code actually has to be maintained XD

Reply:
yeah I understand

Reply:
still missing refactor...